src/llmcompressor/modifiers/utils/hooks.py

```diff
@@ -29,18 +29,29 @@ class HooksMixin(BaseModel):
         - modifier.remove_hooks()
     """
 
-    _HOOKS_DISABLED: ClassVar[bool] = False  # attached to global HooksMixin
-    _hooks: Set[RemovableHandle] = set()  # attached to local subclasses
+    # attached to global HooksMixin class
+    _HOOKS_DISABLED: ClassVar[bool] = False
+    _HOOKS_KEEP_ENABLED: ClassVar[Set[RemovableHandle]] = set()
+
+    # attached to local subclasses
+    _hooks: Set[RemovableHandle] = set()
 
     @classmethod
     @contextlib.contextmanager
-    def disable_hooks(cls):
-        """Disable all hooks across all modifiers"""
+    def disable_hooks(cls, keep: Set[RemovableHandle] = set()):
+        """
+        Disable all hooks across all modifiers. Composing multiple contexts is
+        equivalent to the union of `keep` arguments
+
+        :param keep: optional set of handles to keep enabled
+        """
         try:
             cls._HOOKS_DISABLED = True
+            cls._HOOKS_KEEP_ENABLED |= keep
```

Comment:
use a better variable name

Reply:
Any suggestions? I think this name perfectly describes the set as "the set of hooks which we will keep enabled"

Reply:
On a second reading, that makes sense.
HOOKS_TO_KEEP_ENABLED or HOOKS_KEPT_ENABLED

Reply:
I like the current name

Reply:
I agree it is a little non-descript as is, `HOOKS_TO_KEEP_ENABLED` seems like a minimally invasive update. `HOOKS_KEEP_ENABLED` sounds like a bool type.

None of this will persist in a config/save file right? In that case we don't really have to worry about whether that is the best long-term way to retain state.